hyperspy/drawing/utils.py

```
@@ -1035,9 +1040,12 @@ def transparent_single_color_cmap(color):
     if overlay:
         # Check if images all have same scale and therefore can be overlayed.
         for im in images:
-            if im.axes_manager[0].scale != images[0].axes_manager[0].scale:
+            # tolerate up to 1% relative difference
```

Wouldn't it be better to raise an error, explaining the issue and how to fix it in case that the user is unaware of the mismatch and its cause?

---

I have changed to use the quantity to cover the case, where the units and scale could be different, but the quantity still the same.
I also changed to the same approach as in https://github.com/hyperspy/rosettasciio/pull/414 but this time with quantity. If it fails, it is most likely that the scalebar is not valid on this data.

---

Won't you consider adding a warning to make the user aware of the mismatch?

---

Should we leave it as it is, following the same logic as in https://github.com/hyperspy/rosettasciio/pull/414?